Platform/Ampere/buildfw.sh

```diff
@@ -0,0 +1,287 @@
+ROOT_DIR=$(realpath ${SCRIPT_DIR}/../../..)
+trap ctrl_c INT
+pushd "${ROOT_DIR}"
+
+SPI_SIZE_MB=32
+
+MANUFACTURER=Ampere
+BOARD_NAME=Jade
+SCRIPTS_DIR=$PWD/edk2-ampere-tools/
+
+FLASHFW=0
+RESET_NV_STORAGE=0
+TOOLCHAIN=GCC
+BLDTYPE=RELEASE
+BUILD_THREADS=$(getconf _NPROCESSORS_ONLN)
+
+export PYTHON_COMMAND=python3
+export WORKSPACE=$PWD
+
+if [ "$(uname -o)" = "FreeBSD" ]; then
+  MAKE_COMMAND=gmake
+  GETOPT_COMMAND=/usr/local/bin/getopt
+  if ! command -v ${GETOPT_COMMAND} >/dev/null 2>&1; then
+    echo "GNU getopt is required. Please install the getopt package."
+    exit 1
+  fi
+  if ! command -v ${MAKE_COMMAND} >/dev/null 2>&1; then
+    echo "GNU make is required. Please install the gmake package."
+    exit 1
+  fi
+  mkdir bin || true
+  ln -sfv /usr/local/bin/gmake bin/make
+  export PATH=$PWD/bin:$PATH
+else
+  MAKE_COMMAND=make
+  GETOPT_COMMAND=getopt
+fi
+
+if ! command -v cert_create >/dev/null 2>&1; then
+  tfa_usage
+fi
+
+if ! command -v fiptool >/dev/null 2>&1; then
+  tfa_usage
+fi
+
+if ! command -v python3 >/dev/null 2>&1; then
+  echo "Could not find python3. Please install the python3 package."
+  exit 1
+fi
+
+OPTIONS=$(${GETOPT_COMMAND} -o t:b:fFhl:m:p: --long toolchain:,build:,linuxboot:,manufacturer:,platform:,flash,full-flash,help -- "$@")
+eval set -- "$OPTIONS"
+
+while true; do
+  case "$1" in
+    -t|--toolchain)
+      TOOLCHAIN=$2; shift 2;;
+    -b|--build)
+      BLDTYPE=$2; shift 2;;
+    -f|--flash)
+      FLASHFW=1; shift;;
+    -F|--full-flash)
+      FLASHFW=1; RESET_NV_STORAGE=1; shift;;
+    -l|--linuxboot)
+      LINUXBOOT=LinuxBoot; FLASHKERNEL=$2; shift 2;;
+    -m|--manufacturer)
+      MANUFACTURER=$2; shift 2;;
+    -p|--platform)
+      BOARD_NAME=$2;  shift 2;;
+    -h|--help)
+      usage; shift;;
+    --) shift; break;;
+    *) echo "Internal error ($1)!"; exit 1;;
+  esac
+done
+
+eval set -- ""
+
+BOARD_SETTINGS_CFG=edk2-platforms/Platform/${MANUFACTURER}/${BOARD_NAME}Pkg/${BOARD_NAME}BoardSetting.cfg
+OUTPUT_BIN_DIR=$PWD/Build/${BOARD_NAME}
+OUTPUT_BOARD_SETTINGS_BIN=${OUTPUT_BIN_DIR}/$(basename ${BOARD_SETTINGS_CFG}).bin
+
+export PACKAGES_PATH=$PWD:$PWD/edk2:$PWD/edk2-non-osi:$PWD/edk2-platforms:$PWD/edk2-platforms/Platform/${MANUFACTURER}/${BOARD_NAME}Pkg:$PWD/edk2-platforms/Features/Intel/Debugging:$PWD/edk2-platforms/Features:$PWD/edk2-platforms/Features/Intel
+
+case $(uname -m) in
+  "x86_64")
+    if [ "$TOOLCHAIN" = "GCC" -a -z "${GCC_AARCH64_PREFIX}" ]; then
+      echo "Error: need to define \$GCC_AARCH64_PREFIX since the native compiler won't work"
+      exit 1
+    fi
+    ;;
+esac
+
+if [ "${TOOLCHAIN}" = "CLANG" ]; then
+  TOOLCHAIN=CLANGDWARF
+fi
+
+if [ -n "${LINUXBOOT}" ]; then
+  if [ ! -f "${FLASHKERNEL}" ]; then
+    echo "LinuxBoot flashkernel file \"${FLASHKERNEL}\" does not exist."
+    exit 1
+  else
+    cp -vf "${FLASHKERNEL}" "${WORKSPACE}/edk2-platforms/Platform/Ampere/LinuxBootPkg/AArch64/flashkernel"
```

Comment:
The LinuxBoot flashkernel path has been changed. Now we will need to specify `LINUXBOOT_FILE` in the build command.